### PR TITLE
Standalone NTP unit tests

### PR DESCRIPTION
**Pull request by polybassa**
Co-authored-by: Pierre Lalet <pierre@droids-corp.org>, <pierre.lalet@cea.fr>
Co-authored-by: Pierre Lorinquer <pierre.lorinquer@ssi.gouv.fr>
Co-authored-by: Neurocinetics <0901653@student.hr.nl>
Co-authored-by: Gabriel Potter <gabriel@potter.fr>
Co-authored-by: Guillaume Valadon <guillaume.valadon@netatmo.com>

### CODE REVIEW

**Comment by guedou** [comment]
@polybassa please sync with me and wait for my Ok before pushing more of these PR. We are already working on similar files =/

**Comment by polybassa** [comment]
@guedou Alright. Sorry. I will wait for your OK.

**Comment by guedou** [comment]
No problem. We will likely run into conflicts nightmares if we rush in pushing this PR too fast =)

**Comment by polybassa** [comment]
Probably. I tried to find pretty much isolated sections in regression.uts. Therefore I have chosen random and ntp.

**Comment by guedou** [comment]
`regression.uts` contains other parted related to NTP =/

**Comment by polybassa** [comment]
Oh... should I update right now? Or should I better wait until you finished your PR?

**Comment by polybassa** [comment]
I've updated.

File: test/regression.uts [hunk 1]
```diff
@@ -951,20 +951,13 @@ assert r.ttl != z
 fuzz(ARP(pdst="127.0.0.1"))
 fuzz(IP()/ARP(pdst='10.0.0.254'))
 
-= fuzz on packets with advanced RandNum
```
Comment (guedou) [inline test/regression.uts:954]: IMHO, this should stay in `regression.uts` as it is not a NTP specific test.

File: test/regression.uts [hunk 2]
```diff
@@ -951,20 +951,13 @@ assert r.ttl != z
...
-x = IP(dst="8.8.8.8")/fuzz(UDP()/NTP(version=4))
-x.show2()
-x = IP(raw(x))
-assert NTP in x
 
 = fuzz on packets with FlagsField
 assert isinstance(fuzz(TCP()).flags, VolatileValue)
 
 = Building some packets
-~ basic IP TCP UDP NTP LLC SNAP Dot11
```
Comment (guedou) [inline test/regression.uts:965]: IMHO, this should stay in `regression.uts` as it is not a NTP specific test.

File: test/regression.uts [hunk 3]
```diff
@@ -951,20 +951,13 @@ assert r.ttl != z
...
+~ basic IP TCP UDP LLC SNAP Dot11
 IP()/TCP()
-Ether()/IP()/UDP()/NTP()
```
Comment (guedou) [inline test/regression.uts:967]: IMHO, this should stay in `regression.uts` as it is not a NTP specific test.

File: test/regression.uts [hunk 4]
```diff
@@ -992,14 +985,6 @@ r
 r == 6
 
 
-= sprintf() function
```
Comment (guedou) [inline test/regression.uts:995]: IMHO, this should stay in `regression.uts` as it is not a NTP specific test.

**Review by guedou — CHANGES_REQUESTED** [review]
I made some comments about that should stay in regression.uts. The other changes are good to me.

**Comment by polybassa** [comment]
Hi, I changed everything back, except the `specific haslayer and getlayer implementations for NTP` test

**Comment by guedou** [comment]
Looks good to me! Could you split VXLAN?

**Comment by polybassa** [comment]
Yes. VXLAN is here: #2913

**Comment by guedou** [comment]
Sorry I meant VRRP.

**Comment by guedou** [comment]
LGTM. Could you rebase againt master?

**Comment by guedou** [comment]
Could you do TFTP next?

**Comment by polybassa** [comment]
Yes, I will do tftp next. I will rebase later, today. 
-- 
Viele Grüße

Nils Weiß

**Comment by polybassa** [comment]
Done

**Pull request merged**